src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/OxoGReadCounts.java

```diff
@@ -81,6 +84,155 @@ public void annotate(final ReferenceContext refContext,
         gb.attribute(GATKVCFConstants.F2R1_KEY, f2r1);
     }
 
+    /**
+     *  Annotate the given variant context with the OxoG read count attributes, directly using the read pileup.
+     *
+     *  This method may be slow and should be considered EXPERIMENTAL, especially with regard to indels and complex/mixed
+     *    variants.
+     *
+     * @param vc variant context for the genotype.  Necessary so that we can see all alleles.
+     * @param gb genotype builder to put the annotations into.
+     * @param readPileup pileup of the reads at this vc.  Please note that this pileup does not have to match the
+     *                   genotype.  In other words, this tool does not check that the pileup was generated from the
+     *                   genotype sample.
+     */
+    public static void annotateSingleVariant(final VariantContext vc, final GenotypeBuilder gb,
+                                             final ReadPileup readPileup, double meanBaseQualityCutoff) {
+        Utils.nonNull(gb, "gb is null");
+        Utils.nonNull(vc, "vc is null");
+
+        // Create a list of unique alleles
+        final List<Allele> variantAllelesWithDupes = vc.getAlleles();
+        final Set<Allele> alleleSet = new LinkedHashSet<>(variantAllelesWithDupes);
+        final List<Allele> variantAlleles = new ArrayList<>(alleleSet);
+
+        // Initialize the mappings
+        final Map<Allele, MutableInt> f1r2Counts = variantAlleles.stream()
+                .collect(Collectors.toMap(a -> a, a -> new MutableInt(0)));
+
+        final Map<Allele, MutableInt> f2r1Counts = variantAlleles.stream()
+                .collect(Collectors.toMap(a -> a, a -> new MutableInt(0)));
+
+        final List<Allele> referenceAlleles = variantAlleles.stream().filter(a -> a.isReference() && !a.isSymbolic()).collect(Collectors.toList());
+        final List<Allele> altAlleles = variantAlleles.stream().filter(a -> a.isNonReference() && !a.isSymbolic()).collect(Collectors.toList());
+
+        if (referenceAlleles.size() != 1) {
+            logger.warn("Number of reference alleles does not equal  for VC: " + vc);
+        }
+
+        // We MUST have  exactly 1 non-symbolic reference allele, a read pileup,
```

Comment:
dangling comma -- were you planning on writing something else?

Reply:
extra space between "have" and "exactly"

Reply:
Done and done.  Not planning on a bigger list.